bot.py

```diff
@@ -251,119 +292,102 @@ def get_members_by_role(role):
+        if role == "shitposting":
+            role = "shitty_people"
+        if role == "promote_a_stream":
+            role = "creeps"
 
-        if game.upper() in valid_games:
-            log.info("[{user}] requested to join game: {game}".format(user=member.name, game=game))
+        if role.upper() in valid_games or role.lower() in valid_roles:
+            log.info("[{user}] requested to join game/role: {role}".format(user=member.name, role=role))
 
             # Check to see if the user already has this role
             for author_role in member.roles:
-                if author_role.name == game:
+                if author_role.name == role:
                     # They did, let them know they already had it
-                    msg = "{user} you are already signed in for {game}."
-                    await client.send_message(message.channel, msg.format(user=member.mention, game=game))
+                    msg = "{user} you are already a member of {role}."
+                    await client.send_message(message.channel, msg.format(user=member.mention, role=role))
                     log.info("[{0}] Role already assigned".format(member))
                     break
             else:
-                role = get_role(game)
+                role = get_role(role)
                 if not role:
                     return False
                 # They didn't have the role, so add it
                 await client.add_roles(member, role)
                 log.info("[{0}] Role added".format(member))
-                reply = "Hello {user}, you are now signed up for {game}. People can tag you instead of EVERYONE by using `@{game}`.\n" \
-                    "You can use `!leave {game}` to be removed from the list at any time."
-                await client.send_message(message.channel, reply.format(user=member.mention, game=game))
+                if role.name in valid_games:
+                    reply = "Hello {user}, you are now signed up for {role}. People can tag you instead of EVERYONE by using `@{game}`.\n" \
+                        "You can use `!leave {game}` to be removed from the list at any time.".format(user=member.mention, role=role)
+                    notification_msg = "{user} has signed up to play {role}! There are currently {players} players available."
+                    await client.send_message(get_channel("looking_for_group"), notification_msg.format(user=member.mention, role=role, players=len(get_members_by_role(role))))
+                elif role.name == "shitty_people":
+                    reply = "Hello {user}, you are now able to access {channel}. You fucking pervert.".format(user=member.mention, channel=get_channel("nsfw_shitposting").mention)
+                elif role.name == "creeps":
+                    reply = "Ok {user}, you will now be notified in {channel} when a stream goes live. You're a piece of work.".format(user=member.mention, channel=get_channel("promote_a_stream").mention)
```

Comment:
There should be an `else` here with a generic "you have been added to this role" message. So we don't have to add code when adding a new role (e.g. dreamhack).